app/services/validator_check_active_service.rb

```diff
@@ -54,34 +34,33 @@ def too_young?(validator)
               .exists?
   end
 
-  def became_inactive?(validator)
-    if ValidatorHistory.where(account: validator.account).exists?
-      unless acceptable_stake?(validator) && !delinquent?(validator)
-        return true
-      end
-    end
-
-    false
+  def validator_histories_for_validator(validator)
+    ValidatorHistory.where(account: validator.account)
+                    .where("created_at > ?", @delinquent_time.ago)
+                    .order(created_at: :desc).to_a
   end
 
-  def is_rpc?(validator)
-    if validator.created_at < (DateTime.now - @delinquent_time) && \
-       !validator.vote_accounts.exists?
-      return true
-    end
+  def should_be_destroyed?(validator)
+    validator_histories_for_validator(validator).empty? ? true : false
+  end
 
+  def should_be_active?(validator)
+    return true if acceptable_stake?(validator) && !long_time_delinquent?(validator)
     false
   end
 
-  def update_scorable(validator)
-    if !too_young?(validator) && became_inactive?(validator)
-      validator.update(is_active: false)
-    end
+  def should_be_rpc?(validator)
+    !validator.vote_accounts.exists?
+  end
 
-    validator.update(is_rpc: true) if is_rpc?(validator)
+  def update_scorable(validator)
+    validator.assign_attributes(
+      is_active: should_be_active?(validator),
+      is_rpc: should_be_rpc?(validator)
+    )
   end
 
-  def delinquent?(validator)
+  def long_time_delinquent?(validator)
     return false if !validator.delinquent?
 
     !non_delinquent_history_exists?(validator)
```

Comment:
Same here, I'd use validator_histories_for_validator method explicitly here, instead of extracting it to another method, because both `non_delinquent_history_exists` and `should_be_destroyed` methods don't add any extra logic at all, they just call another method, and nothing else. So there's no reason for them to exist.